xds/internal/xdsclient/watchers.go

```diff
@@ -28,7 +28,7 @@ import (
 // after the watcher is canceled. The caller needs to handle this case.
 func (c *clientImpl) WatchListener(serviceName string, cb func(xdsresource.ListenerUpdate, error)) (cancel func()) {
 	n := xdsresource.ParseName(serviceName)
-	a, unref, err := c.findAuthority(n)
+	a, unref, err := c.findAuthorityLocked(n)
```

Comment:
Lock isn't held?  And below?

Reply:
Ohh, the other two actually says `must not hold mu`.....

Reply:
Fixed

Reply:
Geez, who wrote this code?  😉